### ISSUE TITLE
Serialize tablet repair and tablet migration

### ISSUE THREAD

**Issue by bhalevy** [scylladb/scylladb#18561]
On the way to fully-automatic tablet repair (https://github.com/scylladb/scylladb/issues/17852), we should serialize tablet repair and tablet migration to prevent races between them.
For example:
- If repair starts during tablet migration, the destination replica will not include sstables written by repair, conflicting with https://github.com/scylladb/scylladb/issues/17507 / https://github.com/scylladb/scylladb/pull/18211
- With dynamic allocation of storage groups (https://github.com/scylladb/scylladb/issues/16877 / https://github.com/scylladb/scylladb/pull/17664), repair must synchronize with tablet cleanup so it won't try to add an sstable to the tablet after the cleanup phase.

This issue calls for using the legacy repair api in 6.0, but using the tablet scheduler to orchestrate "repair" tablet transitions that will be serialized with any other transitions (like migration, but also others, like rebuild).

**Comment by bhalevy** [comment]
Cc @tgrabiec 
Is there anything missing or incorrect in the above?

**Comment by mykaul** [comment]
@bhalevy - this is marked as a blocker for 6.0 - who's actively working on resolving it?

**Comment by bhalevy** [comment]
> @bhalevy - this is marked as a blocker for 6.0 - who's actively working on resolving it?

@tgrabiec already started working on it.
@tgrabiec, do you have a draft PR already?

**Comment by tgrabiec** [comment]
Yes, there is a PR ready: https://github.com/scylladb/scylladb/pull/18641
